conans/client/api/conan_api.py

```diff
@@ -348,23 +360,34 @@ def get_package_ids(self, reference, remote=None):
                         "d5f16437dd4989cc688211b95c24525589acaafd": {
                             "settings": {"compiler": "apple-clang",...},
                             "options": {'options': {'shared': 'False',...}},
-                            "requires": ['mylib/1.0.8:3df6ebb8a308d309e882b21988fd9ea103560e16',...]
+                            "requires": ['mylib/1.0.8#3df6ebb8a308d309e882b21988fd9ea103560e16',...]
                         }
-                    }
+                    },
+                    "error": None
                   }
         """
-        ref = ConanFileReference.loads(reference)
+        try:
+            ref = ConanFileReference.loads(reference)
+        except (ConanException, InvalidNameException):
+            raise ConanException(f"{reference} is not a valid recipe reference, provide a reference"
+                                 f" in the form name/version[@user/channel][#RECIPE_REVISION]")
+        error = None
+        packages_props = {}
         if remote:
             rrev = ref if ref.revision else self.app.remote_manager.get_latest_recipe_revision(ref,
                                                                                                remote)
             try:
                 packages_props = self.app.remote_manager.search_packages(remote, rrev, None)
-            except NotFoundException:
-                # This exception must be catched manually due to a server inconsistency:
+            except (NotFoundException, PackageNotFoundException):
+                # This exception must be caught manually due to a server inconsistency:
                 # Artifactory API returns an empty result if the recipe doesn't exist, but
                 # Conan Server returns a 404. This probably should be fixed server side,
                 # but in the meantime we must handle it here
-                packages_props = {}
+                pass
+            except ConanConnectionError:
+                error = "There was a connection problem."
```

Comment:
This is discarding possibly interesting information about the connection error (timeout, auth, connection reset, network down...)

Reply:
You're completely right. I have to perform some new tests so that's the reason I did not realize it.